staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -171,6 +175,11 @@ type handlerChainConfig struct {
 	proxyMux        *http.ServeMux
 	serviceLister   v1listers.ServiceLister
 	endpointsLister v1listers.EndpointsLister
+
+	// postProxyHandler keeps track of the handler that comes after the proxy so that it can be used
+	// to satisfy delegation cases.  It is set as a side-effect while building the handler chain.
+	// TODO refactor to run the proxy *after* authorization so we can simply run at the end of the handling chain
+	postProxyHandler http.Handler
```

Comment:
I found the name `postProxyHandler` confusing, since it actually runs *instead of* proxying (and at first it made me think the handler would handle a request after it had been proxied). `localHandler` or `localDelegate` or similar might be easier to follow